Scripts/datatypes/purpose.py

```diff
@@ -214,43 +214,34 @@ def __init__(self, specification, zone_data, resultdata):
         self.mapping = self.zone_data.aggregations.mappings[
             param.purpose_matrix_aggregation_level]
         self.aggregates = {}
-        self.own_zone_demand = {}
+        self.within_zone_tours = {}
         self.sec_dest_purpose = None
 
     @property
     def dist(self):
         return self.distance[self.bounds, self.dest_interval]
-
-    def print_data(self):
-        self.resultdata.print_data(
-            pandas.Series(
-                sum(self.generated_tours.values()), self.zone_numbers,
-                name=self.name),
-            "generation.txt")
-        self.resultdata.print_data(
-            pandas.Series(
-                sum(self.attracted_tours.values()),
-                self.zone_data.zone_numbers, name=self.name),
-            "attraction.txt")
-        demsums = {mode: self.generated_tours[mode].sum()
-            for mode in self.modes}
-        demand_all = float(sum(demsums.values()))
-        mode_shares = pandas.concat(
-            {self.name: pandas.Series(
-                {mode: demsums[mode] / demand_all for mode in demsums},
-                name="mode_share")},
-            names=["purpose", "mode"])
-        self.resultdata.print_concat(mode_shares, "mode_share.txt")
-        self.resultdata.print_concat(
-            pandas.concat(
-                {m: self.histograms[m].histogram for m in self.histograms},
-                names=["mode", "purpose", "interval"]),
-            "trip_lengths.txt")
-        self.resultdata.print_matrices(
-            self.aggregates, "aggregated_demand", self.name)
-        for mode in self.aggregates:
-            self.resultdata.print_data(
-                self.own_zone_demand[mode], "own_zone_demand.txt")
+    
+    @property
+    def generated_tours_all(self):
+        return pandas.Series(
+            sum(self.generated_tours.values()), self.zone_numbers, name=self.name)
```

Comment:
I like the idea of feeding these `Series` object as a dict to `ResultData.print_data()` as you have done in `ModelSystem`. This means that the `Series.name` variable does not have any function anymore though. Should we remove it or could it be useful for something else in the future?